include/range/v3/action/insert.hpp

```diff
@@ -76,23 +76,76 @@ namespace ranges
                 return unwrap_reference(cont).insert(p, n, std::forward<T>(t));
             }
 
+            namespace impl {
+                template<typename C, typename I, typename S>
+                using ReserveConcept =
+                    meta::fast_and<
+                        RandomAccessReservable<C>,
+                        SizedIteratorRange<I, S>>;
+
+                template<typename Cont, typename P, typename I, typename S,
+                    typename C = common_iterator<I, S>,
+                    CONCEPT_REQUIRES_(LvalueContainerLike<Cont>() && Iterator<P>() && IteratorRange<I, S>())>
+                auto insert(Cont && cont, P p, I i, S j, std::false_type) ->
+                    decltype(unwrap_reference(cont).insert(p, C{i}, C{j}))
```

Comment:
`insert` is a customization point. If these aren't hooking the customization point, they should be called something else, like `insert_` or `insert_impl`.